core/include/detray/builders/cylinder_portal_generator.hpp

```diff
@@ -110,15 +110,16 @@ template <typename detector_t>
 class cylinder_portal_generator final
     : public surface_factory_interface<detector_t> {
 
-    using scalar_t = typename detector_t::scalar_type;
-    using point3_t = typename detector_t::point3_type;
-    using vector3_t = typename detector_t::vector3_type;
-    using transform3_t = typename detector_t::transform3_type;
+    using algebra_t = typename detector_t::algebra_type;
+    using scalar_t = dscalar<algebra_t>;
+    using point3_t = dpoint3D<algebra_t>;
+    using vector3_t = dvector3D<algebra_t>;
+    using transform3_t = dtransform3D<algebra_t>;
```

Comment:
For instance in this case. The algebra is taken from `detector_t`. Then why not keep the rest of the code as-is? What's the upside of using the traits here, and not just inheriting the types from `detector_t` as we currently do?

Reply:
I wanted to couple surrounding code as little as possible to the local typedefs in the detector. Best case scenario in my opinion would be, to have the detector specify the algebra type and any dependent class takes from that what it needs, thus minimizing the requirements of what needs to be in the detector. 
But I had to realize that this falls apart at several places, where that would be clumsy. I still left these changes in, because it is the more concise way of writing these types.